server/app/views/admin/programs/ProgramImageView.java

```diff
@@ -135,7 +146,20 @@ public Content render(
     DivTag formsAndCurrentCardContainer =
         div().withClasses("grid", "grid-cols-2", "gap-10", "w-full");
     formsAndCurrentCardContainer.with(formsContainer);
-    formsAndCurrentCardContainer.with(renderCurrentProgramCard(request, programDefinition));
+
+    if (settingsManifest.getNorthStarApplicantUi(request)) {
+      DivTag cardPreview;
+      try {
+        String content =
+            northStarProgramCardPreviewController.cardPreview(request, programDefinition.id());
+        cardPreview = div(rawHtml(content));
+      } catch (InterruptedException | ExecutionException e) {
+        cardPreview = div(text("Error generating card preview: " + e.getLocalizedMessage()));
+      }
```

Comment:
(blocking) Make sure the exception gets logged, but don't include the exception details on the screen just in case somewhere in the chain there is something that shouldn't be displayed.